### PR TITLE
Create ESC_calibration.cpp

### LINKED ISSUE #95: ESC calibration
Create a function for calibrating the ESC's using the ESP32.

Include a readme explaining how to carry out the calibration

### PR DESCRIPTION
**Pull request by Runway27**
## Main Changes
This code is for calibrating the ESC
To calibrate the ESC copy and paste this into the `main.cpp` file and flash it to the ESP
## Associated Issues

- Closes #95 

## Tests

This has been tested as described above.

### CODE REVIEW

**Review by jamie-54 — APPROVED** [review]
Looks good to me

**Review by alphaville — CHANGES_REQUESTED** [review]
@Runway27 can you write some comments to explain what this test does? Also edit [this README file](https://github.com/QUB-ASL/bzzz/blob/main/hardware/tests/README.md) and provide a description of the test too.

**Comment by jamie-54** [comment]
@Runway27 update [this README](https://github.com/QUB-ASL/bzzz/tree/feature/95-ESC-calibration/hardware#readme) please and thank you

File: hardware/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+The signal to the ESC is set high before the power is provided to the ESC. 
+This sets it to calibration mode.
+After a particular delay the signal to the ESC is set back to low which completes the calibration. 
+
+### Step to calibrate ESC
+* Take the code from `ESC_calibration.cpp` and paste it to the `main.cpp`
+* Flash it to the ESP32
+* Power on the ESC
+* 2 short beeps would comfirm that the ESC is in calibration mode
+* 3 shot beeps and 1 long beep wou;ld confirm that the ESc is calibrated.
```
Comment (alphaville) [inline hardware/README.md:16]: - shot -> short
- wou;ld confirm -> confirm
- ESc -> ESC

**Comment by alphaville** [comment]
@Runway27 avoid editing the code directly on GitHub; install git on your computer and use the git client instead

File: hardware/ESC_calibration.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#include <Arduino.h>
+#include "motors.hpp"
+ 
+bzzz::MotorDriver motorDriver;
+// sets the delay value to 5 sec
+#define PERIOD_1 5000. 
+void setup()
+{
+    motorDriver.attachEscToPwmPin();
+    motorDriver.writeSpeedToEsc(2000, 2000, 2000, 2000); // Sets the signal to ESC as high
```
Comment (alphaville) [inline hardware/ESC_calibration.cpp:11]: Avoid hard-coded values in the code

**Comment by Runway27** [comment]
> @Runway27 avoid editing the code directly on GitHub; install git on your computer and use the git client instead

For some reason I couldn't push to github on VS code.

File: hardware/ESC_calibration.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include <Arduino.h>
+#include "motors.hpp"
+ 
+bzzz::MotorDriver motorDriver;
+// sets the delay value to 5 sec
+#define DELAY_BETWEEN_HIGH_LOW_SIGNALS 5000.
+// Defining the signal high parameter
+#define High_Sgn 2000.
```
Comment (alphaville) [inline hardware/ESC_calibration.cpp:8]: Such variables are `ALL_CAPS`

**Review by alphaville — APPROVED** [review]

**Pull request merged**